### DISCUSSION TITLE
Is the anonymous authorization back-up required when using the DeDRM plugin?

### DISCUSSION THREAD

**Discussion by mtsknn** [Leseratte10/acsm-calibre-plugin#52] [Q&A] [answered]
I'm referring to this bullet point in the readme:

> - You can click on "Create anonymous authorization" to create an anonymous authorization. Make sure to create backups of that authorization.

And these:

> - I suggest using the DeDRM plugin to make sure that losing your AdobeID doesn't also mean you'll lose access to all your eBooks.
> - If you use an anonymous authorization, make sure you make backups of the activation data.

So, if I use anonymous authorization and the DeDRM plugin, I end up with a DRM-free EPUB file. In what cases would I still need the authorization back-up?

Btw thanks for this great plugin! 🙏

**Comment by Leseratte10** [comment, marked as answer]
That's more a general point since it's better to have a backup and not need it, than to need a backup and not have it. In general, if you have the DeDRM plugin installed and active and working, you don't necessarily need to have a backup of your authorization. But if that plugin fails to work for some reason, or is misconfigured, or Adobe releases a new version of the DRM in the future you might have books permanently locked to that anonymous authorization which is not a good thing.

**Reply by mtsknn** [reply]
> That's more a general point since it's better to have a backup and not need it, than to need a backup and not have it.

Great point!

> In general, if you have the DeDRM plugin installed and active and working, you don't necessarily need to have a backup of your authorization.

Nice.

> But if that plugin fails to work for some reason, or is misconfigured, or Adobe releases a new version of the DRM in the future you might have books permanently locked to that anonymous authorization which is not a good thing.

Oh, I see. This risk doesn't apply to any previously successfully de-ACSM'd and de-DRM'd books, right? Because they are already DRM-free EPUB files. But the risk does apply to books I'm adding to Calibre in the future, so I'd better make that back-up, just in case.

Thanks!

**Reply by Leseratte10** [reply]
Correct. The E-Book-Viewer included in Calibre doesn't support any kinds of DRMs. So if you can open an eBook in Calibre's viewer today, that means it's free of DRM and you'll always be able to open this particular book file, independantly from the plugins or from whatever Adobe will do in the future.

**Reply by mtsknn** [reply]
Super, thanks for the detailed replies!

**Comment by wd8rde** [comment]
What directories should be backed up for this? Is it the ".config/calibre/plugins/DeACSM/account" directory?

**Reply by Leseratte10** [reply]
Yes. Or you can just go to the plugin settings and choose "Export account activation data", that'll give you a neat ZIP file with all the account data that you can import again later if needed.

**Comment by natea** [comment]
I've enabled the anonymous authorization and backed it up, but when I try to convert the ACSM file, it says:

```calibre, version 7.26.0
WARNING: Could not convert some books: Could not convert the book because no supported source format was found

Smart Notes Guide - No supported formats (Available formats: acsm)"
```

I've downloaded the DeDRM and DeACSM plugins, restarted Calibre and enabled the plugins. Any ideas why it's not working?

**Reply by Leseratte10** [reply]
The plugin doesn't run upon conversion, it runs upon import. So you need to remove the ACSM from Calibre then import it again and then it should turn into an EPUB or PDF. If it doesn't, you need to restart Calibre in debug mode (right-click the preferences icon then choose restart in debug mode), then try again, then check the log file it shows after exiting.

**Comment by natea** [comment]
In the version of Calibre 8.2.1 that I'm using, I don't see a preferences icon on the toolbar:
![CleanShot 2025-04-10 at 18 37 55](https://github.com/user-attachments/assets/45cd006e-38ce-4039-995e-703a349d42c0)

I do see a preferences option from the menu:
![CleanShot 2025-04-10 at 18 40 38](https://github.com/user-attachments/assets/5456b181-c75e-4189-a880-0be8d88120ab)

But when I open that, I don't see anything resembling a debug mode.

![CleanShot 2025-04-10 at 18 41 35](https://github.com/user-attachments/assets/8632a00e-b453-437c-b791-efc6f3729535)

UPDATE: Okay, now I think I know which debug option you're talking about. When I go into the Plugins screen, and right click on the plugin, then this screen pops up, and from there, I can select the debugging info to a log file.
![CleanShot 2025-04-10 at 18 59 58](https://github.com/user-attachments/assets/dcb6f472-3c83-4db8-945a-797d95021ce1)

(Although it's not clear where I go to look at the debugging log files)

How can I tell if the .epub file that I imported is DRM-free?

**Reply by Leseratte10** [reply]
It looks like Calibre on MacOS handles the preferences icon differently. I don't have access to a Mac so I'm not sure how it works there. 

You can open a Terminal and just type `calibre-debug -g`, that'll also start Calibre in debug mode though it prints the logs to the command line instead of opening a text window afterwards.

The option in the plugin settings is just to make the logs generated in debug mode more verbose / less verbose depending on if it's enabled or not.

> How can I tell if the .epub file that I imported is DRM-free?

If you can open it in the Calibre ebook viewer then it's DRM-free. Calibre only opens books without DRM.
